internal/jobcontainers/jobcontainer.go

```diff
@@ -0,0 +1,429 @@
+		return nil, errors.Wrap(err, "failed to bind payload directory on host")
+	}
+
+	container.sandboxMount = path
+	go container.waitBackground(ctx)
+	return container, nil
+}
+
+// CreateProcess creates a process on the host, starts it, adds it to the containers
+// job object and then waits for exit.
+func (c *JobContainer) CreateProcess(ctx context.Context, config interface{}) (_ cow.Process, err error) {
+	conf, ok := config.(*hcsschema.ProcessParameters)
+	if !ok {
+		return nil, errors.New("unsupported process config passed in")
+	}
+
+	path, args := separateArgs(conf.CommandLine)
+	absPath, err := findExecutable(path, c.sandboxMount)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to find executable")
+	}
+
+	token, err := processToken(conf.User)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create process token")
+	}
+
+	env := append(os.Environ(), envMapToSlice(conf.Environment)...)
+	env = append(env, "CONTAINER_SANDBOX_MOUNT_POINT="+c.sandboxMount)
+	cmd := &exec.Cmd{
+		Env:  env,
+		Dir:  c.sandboxMount,
+		Path: absPath,
+		Args: args,
+		SysProcAttr: &syscall.SysProcAttr{
+			CreationFlags: windows.CREATE_NEW_PROCESS_GROUP,
+			Token:         syscall.Token(token),
+		},
+	}
+
+	process := newProcess(cmd)
+
+	var (
+		stdin  io.WriteCloser
+		stdout io.ReadCloser
+		stderr io.ReadCloser
+	)
+	// Create process pipes if asked for.
+	if conf.CreateStdInPipe {
+		stdin, err = process.cmd.StdinPipe()
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create stdin pipe")
+		}
+		process.stdin = stdin
+	}
+
+	if conf.CreateStdOutPipe {
+		stdout, err = process.cmd.StdoutPipe()
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create stdout pipe")
+		}
+		process.stdout = stdout
+	}
+
+	if conf.CreateStdErrPipe {
+		stderr, err = process.cmd.StderrPipe()
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create stderr pipe")
+		}
+		process.stderr = stderr
+	}
+
+	defer func() {
+		if err != nil {
+			process.Close()
+			c.Release(ctx)
+		}
+	}()
+
+	if err = process.Start(); err != nil {
+		return nil, errors.Wrap(err, "failed to start host process")
+	}
+
+	if err = c.job.Assign(uint32(process.Pid())); err != nil {
+		return nil, errors.Wrap(err, "failed to assign process to job object")
+	}
+
+	// Assign the first process made as the init process of the container.
+	c.init.initDoOnce.Do(func() {
+		c.init.proc = process
+		close(c.init.initBlock)
+	})
+
+	// Wait for process exit
+	go c.pollJobMsgs(ctx)
+	go process.waitBackground(ctx)
+	return process, nil
+}
+
+func (c *JobContainer) Modify(ctx context.Context, config interface{}) (err error) {
+	return errors.New("modify not supported for job containers")
+}
+
+// Release unmounts all of the container layers. Safe to call multiple times, if no storage
+// is mounted this call will just return nil.
+func (c *JobContainer) Release(ctx context.Context) error {
```

Comment:
Is it possible to use the same resource management functions that we have for normal containers here? I know we don't have all of those resources for job containers but still using those functions might be better.

Reply:
Just needed access to where the sandbox is mounted and I didn't want to add a variable to that struct that only applies for job containers (and what you mentioned in that none of the resources would have any purpose here).